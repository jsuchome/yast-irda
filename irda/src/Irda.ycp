/**
 * File:	modules/Irda.ycp
 * Package:	Configuration of irda
 * Summary:	Irda settings, input and output functions
 * Authors:	Jiri Suchomel <jsuchome@suse.cz>
 *
 * $Id$
 *
 * Representation of the IrDA configuration.
 * Input and output routines.
 */

{

module "Irda";
textdomain "irda";

import "FileUtils";
import "Progress";
import "Service";

/**
 * Data was modified?
 */
global boolean modified = false;

/**
 * Should irda be started?
 */
global boolean start	= false;

/**
 * serial port used for irda
 */
global string port	= "";

/**
 * Maximum baud rate for the IrDA serial port
 */
global string max_baud_rate	= "0";

/**
 * If link to /ev/pilot should be created
 */
global boolean link_pilot	= false;

/**
 * Read irda settings from /etc/sysconfig/irda
 * @return true when file exists
 */
global define boolean ReadSysconfig () {


    if (FileUtils::Exists ("/etc/sysconfig/irda"))
    {
	port = (string) SCR::Read (.sysconfig.irda.IRDA_PORT);
	if (port == nil)
	    port = "";

	max_baud_rate = (string) SCR::Read (.sysconfig.irda.IRDA_MAX_BAUD_RATE);
	if (max_baud_rate == nil)
	    max_baud_rate = "0";
	return true;
    }
    return false;
}

/**
 * Read all irda settings
 * @return true on success
 */
global define boolean Read () {

    ReadSysconfig ();

    start = (Service::Status ("irda") == 0);

    return true;
}

/**
 * Write all irda settings
 * @return true on success
 */
global define boolean Write () {

    if (!modified)
	return true;

    /* Irda read dialog caption */
    string caption	= _("Saving IrDA Configuration");
    integer steps	= 2;

    Progress::New (caption, " ", steps, [
	    /* Progress stage 1/2 */
	    _("Write the settings"),
	    /* Progress stage 2/2 */
	    _("Restart the service")
	], [
	    /* Progress step 1/2 */
	    _("Writing the settings..."),
	    /* Progress step 2/2 */
	    _("Restarting service..."),
	    /* Progress finished */
	    _("Finished")
	],
	""
    );

    Progress::NextStage();

    if (port != "")
    {
	SCR::Write (.sysconfig.irda.IRDA_PORT, port);
	SCR::Write (.sysconfig.irda, nil);
    }

    Progress::NextStage ();

    Service::Stop ("irda");
    if (start)
    {
	Service::Start ("irda");
	Service::Enable ("irda");
    }
    else
    {
	Service::Disable ("irda");
    }

    // create symlink to /dev/pilot (#47804)
    if (link_pilot)
    {
	if ((boolean)SCR::Execute (.target.symlink,"/dev/ircomm0","/dev/pilot"))
	{
	    SCR::Execute (.target.bash, "/sbin/resmgr add /dev/pilot desktop");
	}
    }

    Progress::NextStage();

    return true;
}

/* EOF */
}
