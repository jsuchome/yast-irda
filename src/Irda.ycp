/**
 * File:	modules/Irda.ycp
 * Package:	Configuration of irda
 * Summary:	Irda settings, input and output functions
 * Authors:	Jiri Suchomel <jsuchome@suse.cz>
 *
 * $Id$
 *
 * Representation of the IrDA configuration.
 * Input and output routines.
 */

{

module "Irda";
textdomain "irda";

import "Progress";
import "Service";

/**
 * Data was modified?
 */
global boolean modified = false;

/**
 * Should irda be started?
 */
global boolean start	= false;

/**
 * serial port used for irda
 */
global string port	= "";



/**
 * Read irda settings from /etc/sysconfig/irda
 * @return true on success
 */
global define boolean ReadSysconfig () {


    if (SCR::Read (.target.size, "/etc/sysconfig/irda") != -1)
    {
	port = (string) SCR::Read (.sysconfig.irda.IRDA_PORT);
	if (port == nil)
	    port = "";
    }
    return port == "";
}

/**
 * Read all irda settings
 * @return true on success
 */
global define boolean Read () {

    ReadSysconfig ();

    start = (Service::Status ("irda") == 0);

    return true;
}

/**
 * Write all irda settings
 * @return true on success
 */
global define boolean Write () {

    if (!modified)
	return true;

    /* Irda read dialog caption */
    string caption	= _("Saving IrDA Configuration");
    integer steps	= 2;

    Progress::New (caption, " ", steps, [
	    /* Progress stage 1/2 */
	    _("Write the settings"),
	    /* Progress stage 2/2 */
	    _("Restart the service")
	], [
	    /* Progress step 1/2 */
	    _("Writing the settings..."),
	    /* Progress step 2/2 */
	    _("Restarting service..."),
	    /* Progress finished */
	    _("Finished")
	],
	""
    );

    Progress::NextStage();

    if (port != "")
    {
	SCR::Write (.sysconfig.irda.IRDA_PORT, port);
	SCR::Write (.sysconfig.irda, nil);
    }

    Progress::NextStage ();

    Service::Stop ("irda");
    if (start)
    {
	Service::Start ("irda");
	Service::Enable ("irda");
    }
    else
    {
	Service::Disable ("irda");
    }

    // create symlink to /dev/pilot
    if (start && SCR::Read (.target.size, "/dev/pilot") == -1)
    {
	if ((boolean)SCR::Execute (.target.symlink,"/dev/ircomm0","/dev/pilot"))
	{
	    SCR::Execute (.target.bash, "/sbin/resmgr add /dev/pilot desktop");
	}
    }

    Progress::NextStage();

    return true;
}

/* EOF */
}
