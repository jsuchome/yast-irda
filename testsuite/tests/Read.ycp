/**
 *  Read.ycp
 *  Test of Irda::Read function
 *  Author: Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{

    // testedfiles: Irda.ycp Service.ycp
    import "Testsuite";
    import "Irda";

    map READ = $[
	"init": $[
	    "scripts": $[
		"exists"	: true
	    ],
	],
	// no sysconfig file
	"target": $[
	    "stat"	: $[],
	]
    ];

    Testsuite::Dump ("==== reading... ============================");

    Testsuite::Test(``(Irda::Read ()), [ READ, $[], $[] ], 0);

    Testsuite::Dump ("============================================");

    READ = $[
	"init": $[
	    "scripts": $[
		"exists"	: true
	    ],
	],
	"target": $[
	    "stat"	: $[ "isreg" : true ],
	],
	"sysconfig"	: $[
	    "irda"	: $[
		"IRDA_PORT"		: "/dev/ttyS1",
		"IRDA_MAX_BAUD_RATE"	: nil,
	    ],
	],
    ];

    Testsuite::Test(``(Irda::Read ()), [ READ, $[], $[] ], 0);
}

