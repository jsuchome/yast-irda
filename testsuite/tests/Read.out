Dump	==== reading... ============================
Return	true
Dump	============================================
Read	.sysconfig.irda.IRDA_PORT "/dev/ttyS1"
Read	.sysconfig.irda.IRDA_MAX_BAUD_RATE nil
Execute	.target.bash "/sbin/service irda status" $["TERM":"raw"] 0
Return	true
